src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -1632,56 +1564,14 @@ private SyntaxNode WithModifiersInternal(SyntaxNode declaration, DeclarationModi
-                case SyntaxKind.DelegateDeclaration:
-                    return ((DelegateDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.MethodDeclaration:
-                    return ((MethodDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.OperatorDeclaration:
-                    return ((OperatorDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.ConversionOperatorDeclaration:
-                    return ((ConversionOperatorDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.ConstructorDeclaration:
-                    return ((ConstructorDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.DestructorDeclaration:
-                    return ((DestructorDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.FieldDeclaration:
-                    return ((FieldDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.PropertyDeclaration:
-                    return ((PropertyDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.IndexerDeclaration:
-                    return ((IndexerDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.EventFieldDeclaration:
-                    return ((EventFieldDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.EventDeclaration:
-                    return ((EventDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.Parameter:
-                    return ((ParameterSyntax)declaration).Modifiers;
-                case SyntaxKind.LocalDeclarationStatement:
-                    return ((LocalDeclarationStatementSyntax)declaration).Modifiers;
-                case SyntaxKind.VariableDeclaration:
-                case SyntaxKind.VariableDeclarator:
-                    if (declaration.Parent != null)
-                    {
-                        return GetModifierTokens(declaration.Parent);
-                    }
-                    break;
-                case SyntaxKind.GetAccessorDeclaration:
-                case SyntaxKind.SetAccessorDeclaration:
-                case SyntaxKind.AddAccessorDeclaration:
-                case SyntaxKind.RemoveAccessorDeclaration:
-                    return ((AccessorDeclarationSyntax)declaration).Modifiers;
-                case SyntaxKind.LocalFunctionStatement:
-                    return ((LocalFunctionStatementSyntax)declaration).Modifiers;
+                case MemberDeclarationSyntax memberDecl: return memberDecl.Modifiers;
+                case ParameterSyntax parameter: return parameter.Modifiers;
+                case LocalDeclarationStatementSyntax localDecl: return localDecl.Modifiers;
+                case AccessorDeclarationSyntax accessor: return accessor.Modifiers;
+                case VariableDeclarationSyntax varDecl: return GetModifierTokens(varDecl.Parent);
```

Comment:
>VariableDeclarationSyntax [](start = 21, length = 25)

I don't see `VariableDeclarationSyntax` in the old list.
Same for `VariableDeclaratorSyntax`.

Were those bugs? If so, can you add tests? Thanks

Reply:
they were both there.  In the red section:

```c#
                case SyntaxKind.VariableDeclaration:
                case SyntaxKind.VariableDeclarator:
                    if (declaration.Parent != null)
                    {
                        return GetModifierTokens(declaration.Parent);
                    }
                    break;
```

:)